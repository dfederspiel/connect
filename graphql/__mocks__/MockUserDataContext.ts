import { IUserDataContext } from '@lib/auth/UserDataContext';
import { User } from '@prisma/client';

const user = {
  id: 1,
  domain: 'codefly.ninja',
  email: 'david@codefly.ninja',
} as User;

export const MockUserDataContext = {
  createUser: async () => user,
  getById: async () => user,
  getAll: async () => [user],
  getByEmail: async () => user,
  post: async (user: User) => user,
} as IUserDataContext;
