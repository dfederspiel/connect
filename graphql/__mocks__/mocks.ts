import { User } from '@prisma/client';
import { IMocks, MockList } from '@graphql-tools/mock';
import { faker } from '@faker-js/faker';
export interface IConnectMocks extends IMocks {
  User(): User;
}

export const DataMocks = {
  Int: () => faker.datatype.number(100),
  Float: () => faker.datatype.float(10),
  String: () => faker.random.words(5),
  User: () => ({
    id: faker.datatype.number(100),
    domain: faker.internet.domainName(),
    email: faker.internet.email(),
    skills: () => new MockList([0, 12]),
  }),
  Query: () => ({
    users: () => new MockList([0, 12]),
  }),
} as IConnectMocks;
