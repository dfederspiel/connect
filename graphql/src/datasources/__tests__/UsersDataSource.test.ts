import { MockUserDataContext } from '../../../__mocks__/MockUserDataContext';
import UserDataSource from '../UsersDataSource';

describe('the users datasource', () => {
  it('exists', () => {
    const datasource = new UserDataSource(MockUserDataContext);
    expect(datasource).toBeDefined();
  });

  it('can query all users', async () => {
    const datasource = new UserDataSource(MockUserDataContext);
    const users = await datasource.getAll();
    expect(users.length).toBeGreaterThan(0);
  });

  it('can get a user by id', async () => {
    const datasource = new UserDataSource(MockUserDataContext);
    const user = await datasource.getById('1');
    expect(user).toBeDefined();
    expect(user).toMatchSnapshot();
  });
});
