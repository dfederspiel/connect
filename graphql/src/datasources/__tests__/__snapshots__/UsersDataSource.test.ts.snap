// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the users datasource can get a user by id 1`] = `
Object {
  "domain": "codefly.ninja",
  "email": "david@codefly.ninja",
  "id": 1,
}
`;
